subgraphs/enzyme-core/mappings/v4/ExternalPositionManager.ts

```diff
@@ -1763,6 +1774,158 @@ export function handleCallOnExternalPositionExecutedForFund(event: CallOnExterna
+
+      let stakeWiseVaultToken = ensureStakeWiseVaultToken(stakeWiseVault, event.params.externalPosition);
+
+      let wethAsset = ensureAsset(wethTokenAddress);
+
+      let assetAmount = createAssetAmount(wethAsset, amount, denominationAsset, 'stakewise-redeem', event);
+
+      createStakeWiseStakingPositionChange(
+        event.params.externalPosition,
+        'Redeem',
+        stakeWiseVaultToken,
+        assetAmount,
+        shares,
+        vault,
+        event,
+      );
+
+      let stakingPosition = useStakeWiseStakingPosition(event.params.externalPosition.toHex());
+      stakingPosition.stakedEthAmount = stakingPosition.stakedEthAmount.minus(amount);
+      stakingPosition.save();
+    }
+
+    if (actionId == StakeWiseV3StakingPositionActionId.EnterExitQueue) {
+      let decoded = ethereum.decode('(address,uint256)', event.params.actionArgs);
+
+      if (decoded == null) {
+        return;
+      }
+
+      let tuple = decoded.toTuple();
+
+      let stakeWiseVault = tuple[0].toAddress();
+      let shares = toBigDecimal(tuple[1].toBigInt(), 18);
+
+      let stakeWiseV3EthVault = ExternalSdk.bind(stakeWiseVault);
+      let amount = toBigDecimal(stakeWiseV3EthVault.convertToAssets(tuple[1].toBigInt()), 18);
+
+      let stakeWiseVaultToken = ensureStakeWiseVaultToken(stakeWiseVault, event.params.externalPosition);
+
+      let wethAsset = ensureAsset(wethTokenAddress);
+
+      let assetAmount = createAssetAmount(wethAsset, amount, denominationAsset, 'stakewise-enter-exit-queue', event);
+
+      createStakeWiseStakingPositionChange(
+        event.params.externalPosition,
+        'EnterExitQueue',
+        stakeWiseVaultToken,
+        assetAmount,
+        shares,
+        vault,
+        event,
+      );
+
+      let stakingPosition = useStakeWiseStakingPosition(event.params.externalPosition.toHex());
+      stakingPosition.stakedEthAmount = stakingPosition.stakedEthAmount.minus(amount);
+      stakingPosition.save();
+    }
+
+    if (actionId == StakeWiseV3StakingPositionActionId.ClaimExitedAssets) {
+      let decoded = ethereum.decode('(address,uint256)', event.params.actionArgs);
+
+      if (decoded == null) {
+        return;
+      }
+
+      let tuple = decoded.toTuple();
+
+      let stakeWiseVault = tuple[0].toAddress();
+      let positionTicket = tuple[1].toBigInt();
+
+      let stakingPosition = useStakeWiseStakingPosition(event.params.externalPosition.toHex());
+      let stakeWiseVaultToken = ensureStakeWiseVaultToken(stakeWiseVault, event.params.externalPosition);
+
+      let wethAsset = ensureAsset(wethTokenAddress);
+
+      let exitRequestId = stakeWiseStakingExitRequestId(stakingPosition, stakeWiseVaultToken, positionTicket);
+      let exitRequest = useStakeWiseStakingExitRequest(exitRequestId);
+
+      const sharesBI = BigInt.fromString(
+        exitRequest.shares.times(BigDecimal.fromString('1000000000000000000')).toString(),
```

Comment:
could we have a comment (or assignment to the meaningful variable) `1000000000000000000` is 10**18, and it is used for parsing units. Even better would be to have in utils an opposite function to `toBigDecimal` named `fromBigDecimal`

Reply:
Agree to add `fromBigDecimal` function.